Creation of Resume Section

samuelouteda opened the pull request:
Resume Component:
![image](https://user-images.githubusercontent.com/74317636/103239082-ebe06700-4922-11eb-9543-b8b939634905.png)

TimelineMaker Component:
![image](https://user-images.githubusercontent.com/74317636/103239226-437ed280-4923-11eb-8a42-3f9e63c70cb2.png)

SkillMeter Component:
![image](https://user-images.githubusercontent.com/74317636/103239269-63ae9180-4923-11eb-97fb-8be36680e797.png)

Components with real information (not Lorem Ipsum's):
- TimelineMaker (1/3):
![image](https://user-images.githubusercontent.com/74317636/104049016-94f84e80-51ba-11eb-9825-84013785358c.png)

- TimelineMaker (2/3):
![image](https://user-images.githubusercontent.com/74317636/104049074-ad686900-51ba-11eb-8484-28dad5c1e91d.png)

- TimelineMaker (3/3):
![image](https://user-images.githubusercontent.com/74317636/104049104-beb17580-51ba-11eb-9f03-e8e36c8d9428.png)

- SkillMeter:
![image](https://user-images.githubusercontent.com/74317636/103821687-1cbc4c80-5045-11eb-94ab-53f30d2e5a2a.png)

samuelouteda committed:
Creation of Resume Section

Creation of Resume Section Assembling two global and two local components. The globals are SectionTitle and SectionSubtitle; the locals are TimelineMaker, which displays a timeline separated in lapses, where each one has a title, a subtitle, a paragraph and a date; and SkillMeter, which renders a given number of progress bars with an animation. All this components where imported in an index.js file to be exported and used in Resume.jsx

kilordpepo commented on src/sections/Resume/Resume.jsx:
```diff
@@ -0,0 +1,229 @@
+  TimelineMaker,
+  SkillMeter
+} from "./components";
+import { makeStyles } from "@material-ui/core";
```
all static imports or module imports should go first

kilordpepo commented on src/sections/Resume/Resume.jsx:
```diff
@@ -0,0 +1,229 @@
+} from "./components";
+import { makeStyles } from "@material-ui/core";
+
+const useStyles = makeStyles(theme => ({
```
Please create a trello card to move all styles to a separate file. I realized we are taking too much space to declare this in the component file

samuelouteda replied:
Got it!

kilordpepo commented on src/sections/Resume/Resume.jsx:
```diff
@@ -0,0 +1,229 @@
+    height: styles.sizes.globalDimensions.percentageFull,
+    overflowY: "scroll",
+    background: styles.colors.mediumGray,
+    marginLeft: styles.sizes.sidebarDimensions.width,
```
If we are going to use this width for different sections then change this name to something more generic. Please apply this for any similar cases

kilordpepo replied:
You can even use it as global dimensions

kilordpepo commented on src/sections/Resume/Resume.jsx:
```diff
@@ -0,0 +1,229 @@
+  return (
+    <div
+      className={classes.resumeSection}
+      id='resumeSection'
```
change notation to 'resume-section'

kilordpepo commented on src/sections/Resume/Resume.jsx:
```diff
@@ -0,0 +1,229 @@
+              <SectionSubtitle subtitleText='EXPERIENCE' />
+            </div>
+            <TimelineMaker
+              timeline={[
```
Create a trello card so that I can provide this information later

samuelouteda replied:
Ready, sir!

kilordpepo commented on src/sections/Resume/Resume.jsx:
```diff
@@ -0,0 +1,229 @@
+            />
+          </div>
+        </div>
+        <div className={classes.skills}>
```
I see the structures above and here to be somehow similar, a title and then a content. Can this be defined as a generic component and then pass the title and content as a prop? (maybe including some styles)

kilordpepo commented on src/sections/Resume/Resume.jsx:
```diff
@@ -0,0 +1,229 @@
+        </div>
+        <div className={classes.skills}>
+          <div className={classes.skillCards}>
+            <div className={classes.subtitle} id='myBar'>
```
'my-bar'

kilordpepo replied:
better descriptive name

kilordpepo commented on src/sections/Resume/components/SkillMeter/SkillMeter.js:
```diff
@@ -0,0 +1,145 @@
+    }
+  };
+
+  animate();
```
This has to be run every time the component renders? I really don't like this, is there another possible way to make this more efficient?

kilordpepo replied:
Try using useEffect hook or something else that could actually activate this only in certain escenarios

kilordpepo commented on src/sections/Resume/components/SkillMeter/SkillMeter.js:
```diff
@@ -0,0 +1,145 @@
+          key={sk.key}
+          className={classes.skillItem}
+        >
+          <Typography className={classes.skillLabel} variant='subtitle1'>
```
more descriptive variant

kilordpepo commented on src/sections/Resume/components/SkillMeter/SkillMeter.js:
```diff
@@ -0,0 +1,145 @@
+    <div>
+      {skills.map((sk, index) => (
+        <div
+          id={`${sk.key}-progressBar`}
```
change this id and every other id to the correct notation

kilordpepo commented on src/sections/Resume/components/TimelineMaker/TImelineMaker.js:
```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { styles } from "../../../../styles";
+import { makeStyles, Typography } from "@material-ui/core";
```
move this up

kilordpepo commented on src/sections/Resume/components/TimelineMaker/TImelineMaker.js:
```diff
@@ -0,0 +1,146 @@
+    <ul className={classes.timelineWrapper}>
+      {timeline.map(time => (
+        <li
+          key={`${time.date.from}-${time.date.to}`}
```
This key sometimes is not unique

kilordpepo commented on src/sections/Resume/components/TimelineMaker/TImelineMaker.js:
```diff
@@ -0,0 +1,146 @@
+            <span className={`${classes.dateYearMobile} ${classes.dateYearTo}`}>
+              {time.date.to}
+            </span>
+            <div className={classes.datelineCircle}></div>
```
Self closing div? <div/>

kilordpepo commented on src/sections/Resume/components/TimelineMaker/TImelineMaker.js:
```diff
@@ -0,0 +1,146 @@
+            <div className={classes.datelineCircle}></div>
+            <div className={classes.dateline}>
+              <div className={classes.cardContent}>
+                <Typography className={classes.cardTitle} variant='subtitle1'>
```
please change variants to something more descriptive

kilordpepo commented on src/sections/Resume/components/TimelineMaker/TImelineMaker.js:
```diff
@@ -0,0 +1,146 @@
+                </Typography>
+              </div>
+            </div>
+            <div className={classes.datelineCircle}></div>
```
same here self closing tag

kilordpepo reviewed (commented):
Some changes to be needed

samuelouteda committed:
Changing Import's order and some minor changes

Changing import's order in InputTextField, SectionSubtitle, SectionTitle, StyledButton, AboutMe.jsx, ListMaker, ServicesInfo, AssembledForm, ContactInfo, ContactMe, HomeVid, HomeFullName, Home.jsx, SkillMeter, Resume.jsx,  correcting name of TimelineMaker and changing the import on Resume/components/index.js to use the new one, moving key on SideBarLinks and correcting disableTypography's boolean, importing SectionSubtitle on SkillMeter and TimelineMaker and using it to render a title with styles based on a prop called subtitle, adding PropTypes to that two components due to the new received prop, changing ID on SkillMeter to follow the correct syntax, refactoring SkillMeter animation function to use useEffect Hook, changing key on TimelineMaker to be unique always and adding a div to wrap the new Typography element with the old ones, removing the subtitles on Resume.jsx and passing subtitle's prop to SkillMeter and TimelineMaker because they will render the subtitles now, adding key to SideBarFooter, changing sidebarDimension's rules on styles.js to be more generic and applying these changes on SideBar.jsx, AboutMe.jsx, ContactMe.jsx and Resume.jsx

samuelouteda committed:
Saving long styles in separated files and applying minor improvements

Changing helperText's element from a div to a span, removing unnecessary color prop and removing 'isRequired' from error and helperText PropTypes in InputTextField, adding breakpoints to SideBarProfile to ensure that the ProfilePic will always be square, abstracting common styles from AboutMe.jsx, Resume.jsx and ContactMe.jsx and saving them into an object called standardClasses on styles.js to be exported and used where its needed, removing unused fontFamilies from styles.js, importing new standardClasses object in styles/index.js, adding key prop to ContactInfo, correcting props in HomeVid, removing styles from SkillMeter, TimelineMaker and SideBar.jsx due to its length and saving them into files in the same folders, updating animationKey on SkillMeter to stop the animation when the last progress bar is in view, adding profile-pic image file, importing and passing it to ProfilePic on SideBar.jsx to be rendered, changing routes passed to SideBarLinks on SideBar.jsx,  replacing Lorem Ipsum's with real information on Resume.jsx and installing React Router

samuelouteda committed:
Modifying Timeline's dates and styles

Modifying Timeline's dates and styles passed as props from Resume.jsx to TimelineMaker, improving responsive design of Resume.jsx, changing the PropTypes of TimelineMaker to define the new received props and applying these prop's information in the Timeline and changing TimelineMaker's styles.js to style the new information.
